### PR TITLE
feat(server): ability to configure grpc server settings

### LINKED ISSUE #86: ability to configure grpc connection settings
**Summary**

Add ability to configure grpc connection settings like 
- Timeout
- MaxCallRecvMsgSize
- MaxCallSendMsgSize

### PR DESCRIPTION
**Pull request by bsushmith**
closes #86

### CODE REVIEW

File: internal/server/config.go [hunk 1]
```diff
@@ -49,6 +50,16 @@ type Config struct {
 	Auth                       Auth           `mapstructure:"auth"`
 }
 
+type GRPCConfig struct {
+	Port int `mapstructure:"port" default:"8080"`
+	// TimeoutInSeconds is the maximum time in seconds a request can take before being cancelled. Default = 5
+	TimeoutInSeconds int `mapstructure:"timeout_in_seconds" default:"5"`
+	// MaxCallRecvMsgSize is the maximum message size the server can receive. Default = 1 << 25 (32MB)
```
Comment (rahmatrhd) [inline internal/server/config.go:57]:
```suggestion
	// MaxCallRecvMsgSize is the maximum message size the server can receive in bytes. Default = 1 << 25 (32MB)
```

File: internal/server/config.go [hunk 2]
```diff
@@ -49,6 +50,16 @@ type Config struct {
...
+	MaxCallRecvMsgSize int `mapstructure:"max_call_recv_msg_size" default:"33554432"`
+	// MaxCallSendMsgSize is the maximum message size the server can send. Default = 1 << 25 (32MB)
```
Comment (rahmatrhd) [inline internal/server/config.go:59]:
```suggestion
	// MaxCallSendMsgSize is the maximum message size the server can send in bytes. Default = 1 << 25 (32MB)
```

File: internal/server/server.go [hunk 3]
```diff
@@ -168,14 +169,17 @@ func RunServer(config *Config) error {
 			},
 		}),
 	)
-	address := fmt.Sprintf(":%d", config.Port)
+
+	// grpcPort has to be same as config.Port till the time guardian service can support both grpc and http in two different ports
+	grpcPort := config.Port
+	address := fmt.Sprintf(":%d", grpcPort)
```
Comment (rahmatrhd) [inline internal/server/server.go:175]: in this case, can we remove `GRPCConfig.Port` for now since it's not being used anywhere?

Comment (bsushmith) [reply internal/server/server.go:175]: done

**Review by rahmatrhd — APPROVED** [review]

**Pull request merged**